src/SFML/Window/Unix/Display.cpp

```diff
@@ -129,7 +133,8 @@ Atom getAtom(const std::string& name, bool onlyIfExists)
 
     const auto display = openDisplay();
     const Atom atom    = XInternAtom(display.get(), name.c_str(), onlyIfExists ? True : False);
-    atoms[name]        = atom;
+    if (atom)
+        atoms[name] = atom;
 
     return atom;
```

Comment:
Am I seeing this right, that we'd need to error handling in the ClipboardImpl as well?

https://github.com/SFML/SFML/blob/ae87def93e7b0280b5826b0fa5e1a8ef7156e9e7/src/SFML/Window/Unix/ClipboardImpl.cpp#L83-L87

Reply:
Can you clarify this a bit? I'm not sure where you want the error handling?

Reply:
Doesn't this just delay the failure? Now the atoms are set to nullptr and then blindly used later, e.g. here: https://github.com/SFML/SFML/blob/master/src/SFML/Window/Unix/ClipboardImpl.cpp#L134-141

Or does this cause no issue?

Reply:
Oh I see. I think we only need special handling if `onlyIfExists` is `true`. Otherwise `XInternAtom` always returns a valid atom, or you would get X error would would cause the app to exit.

So in `ClipboardImpl.cpp` it looks like `m_utf8String` is the only atom which could be null, and I think it's handled correctly with lots of `!= None` checks.

The original trigger for this patch is because SFML uses `getAtom("UTF8_STRING", true)` and `getAtom("UTF8_STRING", false)` in different places. The first call caches `None` causing the second call to unexpectedly fail.

Reply:
I trust your judgement on this 😄